Firmware/ultralcd.cpp

```diff
@@ -8947,6 +8954,14 @@ void lcd_experimental_toggle()
     eeprom_update_byte((uint8_t *)EEPROM_EXPERIMENTAL_VISIBILITY, oldVal);
 }
 
+void ECool_toggle(){
+    enableECool = ! enableECool;
+    // this is only called when the experimental menu is visible, thus the first condition for enabling of the ECool mode is met in this place
+    eeprom_update_byte((uint8_t *)EEPROM_ECOOL_ENABLE, enableECool ? EEPROM_ECOOL_MAGIC_NUMBER : EEPROM_EMPTY_VALUE);
+}
+bool ECool_get(){
+    return enableECool;
+}
```

Comment:
It is overly complicated and error prone to cache EEPROM in RAM, and it also costs RAM. It is more efficient to just read status from EEPROM inside ECool_get() and use ECool_get() consistently and remove global variable enableECool.

Also at this point enableECool variable is sometimes used directly and sometimes through ECool_get().

If you remove enableECool variable, you can also remove its initialization from setup(), making things less error prone.

Reply:
Also you can call TMC reinit from ECool_toggle(), so change is applied when the state of the variable changes. If experimental menu is not accessible during print, it is safe.